devtools/enforcer-rules/src/main/java/io/quarkus/enforcer/RequiresMinimalDeploymentDependency.java

```diff
@@ -0,0 +1,146 @@
+package io.quarkus.enforcer;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.enforcer.rule.api.EnforcerRule;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleHelper;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluationException;
+
+public class RequiresMinimalDeploymentDependency implements EnforcerRule {
+
+    private static final String GROUP_ID_FILTER = "io.quarkus";
+
+    private static final Map<String, Optional<String>> DEPLOYMENT_GAV_CACHE = new ConcurrentHashMap<>();
+
+    private Log logger;
+
+    @Override
+    public void execute(EnforcerRuleHelper helper) throws EnforcerRuleException {
+        logger = helper.getLog();
+        MavenProject project;
+        try {
+            project = (MavenProject) helper.evaluate("${project}");
+        } catch (ExpressionEvaluationException e) {
+            throw new IllegalStateException(e);
+        }
+
+        // general notes:
+        // - "artifacts" are needed to retrieve the actual jar files
+        // - "dependencies" are needed to limit the scope to only the direct dependencies of the current module
+        //   and to check for the proper scope, type and exclusions
+        // - parents are not gathered here since it is expected that the rule is also active for those parents
+        //   or those parents are not relevant
+
+        Map<String, Dependency> directQuarkusDepsByGAV = project.getDependencies().stream()
+                .filter(d -> d.getGroupId().equals(GROUP_ID_FILTER))
```

Comment:
just a note, I'm wondering if `startsWith` won't be safer here (and in line 55) in case we e.g. wanted to move grpc stuff to `io.quarkus.grpc`

Reply:
Replaced with `startsWith` as suggested.